### PR TITLE
Teleport QoL updates

### PR DESCRIPTION
**Pull request by Michaelzag**
Teleport QoL updates

### CODE REVIEW

**Comment by Michaelzag** [comment]
Working on reducing complexity

**Comment by Michaelzag** [comment]
Pulling out conditionals to separate methods makes the code less readable and there’s no reason for it. The methods are just longer to handle all of the TP logic.

**Comment by Intelli** [comment]
Please refactor utilizing SonarLint & pass CodeFactor. https://github.com/PlayPro/CoreProtect/blob/master/CONTRIBUTING.md

As well, due to the complexity of your changes, please move all your code to a new class, such as TeleportAdvanced, and leave the existing teleport code unmodified.

File: src/main/java/net/coreprotect/model/BlockGroup.java [hunk 1]
```diff
@@ -33,7 +33,8 @@
...
+    public static Set<Material> TARGETABLE_THROUGH = new HashSet<>(Arrays.asList(Material.ACACIA_TRAPDOOR, Material.BIRCH_TRAPDOOR, Material.DARK_OAK_TRAPDOOR, Material.JUNGLE_TRAPDOOR, Material.OAK_TRAPDOOR, Material.SPRUCE_TRAPDOOR, Material.ACACIA_FENCE, Material.ACACIA_FENCE_GATE, Material.BIRCH_FENCE, Material.BIRCH_FENCE_GATE, Material.DARK_OAK_FENCE, Material.DARK_OAK_FENCE_GATE, Material.JUNGLE_FENCE, Material.JUNGLE_FENCE_GATE, Material.OAK_FENCE, Material.OAK_FENCE_GATE, Material.SPRUCE_FENCE, Material.SPRUCE_FENCE_GATE));
```
Comment (Intelli) [inline src/main/java/net/coreprotect/model/BlockGroup.java:37]: Please do not add blockl sets. These are considered deprecated, and are being targeted for future removal.

Please either use existing API methods such as `isSolid`, or use one of the vanilla block tags: https://hub.spigotmc.org/javadocs/bukkit/org/bukkit/Tag.html

**Pull request closed without merging**